### PR TITLE
Introduce some `Map`-usage in the AppOptions

### PR DESCRIPTION
**Pull request by Snuffleupagus**
This simplifies both checking if an entry exists and iteration of the entries, and will help reduce the size/scope of PR #18465.

### CODE REVIEW

File: web/app_options.js [hunk 1]
```diff
@@ -492,7 +498,9 @@ class AppOptions {
   }
 
   static get(name) {
-    return userOptions[name] ?? defaultOptions[name]?.value ?? undefined;
+    return userOptions.has(name)
```
Comment (calixteman) [inline web/app_options.js:501]: Could we avoid `null` or `undefined` values in options and then just avoid `has` & `get` to just use `get` ?

Comment (Snuffleupagus) [reply web/app_options.js:501]: Unfortunately I don't believe that possible in general, hence these changes, given:
 - The values that may be used with browser-options.
 - Upcoming changes in PR #18465.

**Review by timvandermeij — APPROVED** [review]

**Comment by timvandermeij** [comment]
Thank you for the patch!

**Pull request merged**